### DISCUSSION TITLE
JWT access token validation

### DISCUSSION THREAD

**Discussion by kohtala** [panva/openid-client#393] [Ideas]
I have a REST API server where I'd like to accept bearer access tokens in the Authorization header. The client receives the access token from Azure AD.

The [access token from Azure](https://docs.microsoft.com/fi-fi/azure/active-directory/develop/access-tokens#claims-in-access-tokens) is much the same as an id_token received from Azure. Looking at openid-client, the Client.validateJWT seems to be doing almost everything required apart the scp claim unique to the access token. As scp is a proprietary claim, the application can validate that.

But the validateJWT is marked private. The feature I'd like to discus is as simple as if it is possible to mark validateJWT public and include in the typescript.

**Comment by panva** [comment]
I do not intend to expose functionality for resource servers.

**Reply by kohtala** [reply]
Hmm. Bummer.

I also plan to use the Strategy. Very tempting to use same Issuer and key caches for both.

**Reply by kohtala** [reply]
I made a related #394. I wonder if the new package with Strategy will be better place for this.
